bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/TypeConverter.java

```diff
@@ -263,7 +264,44 @@ static boolean isConvertibleToFloatingPointTypes(Object value) {
         return convertibleTypes;
     }
 
-    private static boolean isConvertibleToRecordType(Object sourceValue, BRecordType targetType,
+    public static List<BType> getConvertibleTypesFromJson(Object value, BType targetType,
+                                                          List<TypeValuePair> unresolvedValues) {
+        List<BType> convertibleTypes = new ArrayList<>();
+
+        int targetTypeTag = targetType.getTag();
+
+        convertibleTypes.addAll(TypeConverter.getConvertibleTypes(value, targetType));
+
+        if (convertibleTypes.size() == 0) {
+            switch (targetTypeTag) {
+                case TypeTags.RECORD_TYPE_TAG:
+                    if (isConvertibleToRecordType(value, (BRecordType) targetType, true, unresolvedValues)) {
+                        convertibleTypes.add(targetType);
+                    }
+                    break;
+                case TypeTags.TABLE_TAG:
+                    if (((BTableType) targetType).getConstrainedType().getTag() == TypeTags.RECORD_TYPE_TAG ||
+                            ((BTableType) targetType).getConstrainedType().getTag() == TypeTags.MAP_TAG) {
```

Comment:
Re: the constraint, the spec says

> The type specified by a row-type-parameter must be a subtype of map<any|error>

So it is always a record or a map right? Do we need this check?

Reply:
I think yes. because  If I remove the above-mentioned check, 
```ballerina
public function main() {
    json j = [
        "cake",
        "buscuit"
    ];
    table<string>|error tabString = j.fromJsonWithType(TableString);
}
```
This works without giving a prior error for `table<string>`